src/spmBatching/setBatchRealign.m

```diff
@@ -32,31 +46,68 @@
             [voxDim, opt] = getFuncVoxelDims(opt, subFuncDataDir, prefix, fileName);
 
             if numel(file) > 1
-                errorStruct.identifier = 'setBatchRealignReslice:tooManyFiles';
+                errorStruct.identifier = 'setBatchRealign:tooManyFiles';
                 errorStruct.message = 'This should only get on file.';
                 error(errorStruct);
             end
 
             fprintf(1, ' %s\n', file{1});
 
-            matlabbatch{end}.spm.spatial.realign.estwrite.data{1, runCounter} = file;
+            if strcmp(action, 'realignUnwarp')
+                matlabbatch{end}.spm.spatial.realignunwarp.data(1, runCounter).pmscan = '';
+                matlabbatch{end}.spm.spatial.realignunwarp.data(1, runCounter).scans = file;
+            else
+                matlabbatch{end}.spm.spatial.realign.estwrite.data{1, runCounter} = file;
+            end
 
             runCounter = runCounter + 1;
         end
 
     end
 
+    %% defaults
     % The following lines are commented out because those parameters
     % can be set in the spm_my_defaults.m
+
+    % -----------------
+    % REALIGN
+    % REALIGN & RESLICE
+    % -----------------
+    %
     % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.quality = 1;
     % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.sep = 2;
     % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.fwhm = 5;
-    % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.rtm = 1;
+    % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.rtm = 0;
```

Comment:
@Remi-Gau uhmm I do not know how you use this spm_defaults + this section. Is it for visualisation of what is set in spm_defaults or is it the opposite? Also, I'm a bit lost (not remembering what I was doing previously), what was in the previous form of cpp-pipeline? This was set to 1 and now you are changing it to 0? And how is it in spm_defaults?

Reply:
Yes this is only for "users" to see what the other defaults of the pipelines are.

Currently, the defaults of the pipeline are set in [src/spmDefautls](https://github.com/cpp-lln-lab/CPP_BIDS_SPM_pipeline/tree/dev/src/spmDefaults).

Any users can "override" those by having their own "spm_defaults" in the matlab path.
I actually need to :
- document that better
- make sure that the order in which those `spm_defaults` appear in path is the correct one (to make sure that the user's default takes precedence over those of the pipeline.) 

When in doubt about what SPM uses as defaults, you can type `edit spm_defaults`. :wink: The file also comes with a "how to use this" section. :-) 

So for realign the `realign to mean` (rtm) is set to `true` but it is false `for realign and unwarp`.

I am just setting this to `false` for all the cases of realingment we use.

Does that make sense?